Mathlib/Data/Bitvec/Lemmas.lean

```diff
@@ -124,4 +130,65 @@ theorem ofFin_toFin {n} (v : BitVec n) : ofFin (toFin v) = v := by
   rfl
 #align bitvec.of_fin_to_fin Std.BitVec.ofFin_toFin
 
+/-!
+  ## Distributivity of ofFin
+  We add simp-lemmas that show how `ofFin` distributes over various bitvector operations, showing
+  that bitvector operations are equivalent to `Fin` operations
+-/
+@[simp] lemma neg_ofFin (x : Fin (2^w)) : ofFin (-x) = -(ofFin x) := by
+  rw [neg_eq_zero_sub]; rfl
+
+@[simp] lemma ofFin_and_ofFin (x y : Fin (2^w)) : ofFin (x &&& y) = ofFin x &&& ofFin y := rfl
+@[simp] lemma ofFin_or_ofFin  (x y : Fin (2^w)) : ofFin (x ||| y) = ofFin x ||| ofFin y := rfl
+@[simp] lemma ofFin_xor_ofFin (x y : Fin (2^w)) : ofFin (x ^^^ y) = ofFin x ^^^ ofFin y := rfl
+@[simp] lemma ofFin_add_ofFin (x y : Fin (2^w)) : ofFin (x + y)   = ofFin x + ofFin y   := rfl
+@[simp] lemma ofFin_sub_ofFin (x y : Fin (2^w)) : ofFin (x - y)   = ofFin x - ofFin y   := rfl
+@[simp] lemma ofFin_mul_ofFin (x y : Fin (2^w)) : ofFin (x * y)   = ofFin x * ofFin y   := rfl
+
+/-!
+  ## Distributivity of toFin
+-/
+@[simp] lemma toFin_neg (x : BitVec w) : (-x).toFin = -(x.toFin) := by
+  rw [neg_eq_zero_sub]; rfl
+
+/-!
+## Ring
+-/
+
+lemma zero_eq_ofFin_zero : 0#w = ofFin 0 := rfl
+lemma one_eq_ofFin_one   : 1#w = ofFin 1 := rfl
+
+@[reducible]
+instance : SMul ℕ (BitVec w) where
+  smul x y := x#w * y
+
+@[reducible]
+instance : SMul ℤ (BitVec w) where
+  smul x y := (BitVec.ofInt w x) * y
+
+@[reducible]
+instance : Pow (BitVec w) ℕ where
+  pow :=
+    let rec pow x n :=
+      match n with
+      | 0   => 1
+      | n+1 => x * (pow x n)
+    pow
+
+instance : NatCast (BitVec w) := ⟨BitVec.ofNat w⟩
+instance : IntCast (BitVec w) := ⟨BitVec.ofInt w⟩
+
+-- lemma toFin_nsmul (n : ℕ) (x : BitVec w) :
+--   (n • x).toFin = n
+
+/-! Now we can define an instance of `CommRing (BitVector w)` straightforwardly in terms of the
+    existing instance `CommRing (Fin (2^w))` -/
+instance : CommRing (BitVec w) := by
+  apply Function.Injective.commRing _ toFin_injective <;> (intros; try rfl)
```

Comment:
rather than `intros; try rfl`, can you use the names of the lemmas that you already created above? It's a nice check that you wrote all the ones that matter.